### PR TITLE
The death of numbered vermin

### PR DESCRIPTION
**Pull request by Kaedwuff**
-Removes numerical designations on every mob type I could find
-Implements a new system of randomized descriptive adjectives pulled from lists that can be later used or modified as needed.  This affects rats, drones, golems, contructs, slimes, and several monkey variants.  It's possible I missed things, I can add those later if needed.
-Rats can spawn randomly as vermin, mice, or rodents instead, for additional variety.
-Rats and Maint drones players can edit their name to a simple "[descriptor] [mobtype]".  The game does check to make sure your chosen name is valid before allowing you to take it.  Name changes generate an adminlog, so you will be noticed immediately if you start trying to be a nazi rat or some stupid shit.
-Slime players can now also change their names, with less limitations.
-A namechange helper was created for use of the former system, and can be adapted by anyone for use in a name-change proc with the addition of a regex guideline/definition.  
-Lizards lost their typo-borne proper pronoun status and are now just 'a lizard' instead of "Lizard'.

### CODE REVIEW

**Comment by Kaedwuff** [comment]
!review

File: code/modules/mob/living/simple_animal/friendly/rat.dm [hunk 1]
```diff
@@ -100,6 +100,32 @@
...
+
+	if (M == src)
+		log_and_message_admins(src, "changed their name to [input].")
+	else
+		log_and_message_admins(M, "changed [key_name(src)]'s name to [input].") //To track name changes and ooc shittery.
+	name = input
+	real_name = name
+	
+
+var/regex/ratgex = regex(@"^[a-z]+(?:-[a-z]+?)? (?:rat|rodent|vermin|mouse)$") // making sure names fall within logical parameters
```
Comment (MarinaGryphon) [inline code/modules/mob/living/simple_animal/friendly/rat.dm]: "a vermin", as a phrase, just does not work.
```suggestion
var/regex/ratgex = regex(@"^[a-z]+(?:-[a-z]+?)? (?:rat|rodent|mouse)$") // making sure names fall within logical parameters
```

File: code/modules/mob/living/simple_animal/friendly/rat.dm [hunk 2]
```diff
@@ -108,7 +134,8 @@
 	verbs += /mob/living/proc/hide
 
 	if(name == initial(name))
-		name = "[name] ([rand(1, 1000)])"
+		name = "[pick(vermin_descriptors)] [pick("rat", "rodent", "vermin", "mouse")]"
```
Comment (MarinaGryphon) [inline code/modules/mob/living/simple_animal/friendly/rat.dm]:
```suggestion
		name = "[pick(vermin_descriptors)] [pick("rat", "rodent", "mouse")]"
```
No 'a vermin', please.

File: code/modules/mob/living/simple_animal/friendly/rat.dm [hunk 3]
```diff
@@ -132,6 +159,14 @@
 
 	SSmob.all_rats += src
 
+/mob/living/simple_animal/rat/verb/rename_vermin()
+	set src = usr
+	set name = "Personalize Vermin"
+	set category = "IC"
+	set desc = "Describe your vermin."
+
+	rename_self_helper(usr, ratgex, "How do you want to describe your rodent?\nYour descriptive name needs to be be one word (or hyphenated compound word), followed by one of: rat, vermin, rodent, mouse.", "Your descriptive name needs to be be one word (or hyphenated compound word), then rat/vermin/rodent/mouse.")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/simple_animal/friendly/rat.dm]:
```suggestion
	rename_self_helper(usr, ratgex, "How do you want to describe your rodent?\nYour descriptive name needs to be be one word (or hyphenated compound word), followed by one of: rat, rodent, mouse.", "Your descriptive name needs to be be one word (or hyphenated compound word), then rat/rodent/mouse.")
```

**Comment by Kaedwuff** [comment]
@MarinaGryphon 

There is some intrinsic humor value in being called something like a 'racid vermin', and in this system no rat that spawns will ever be called 'a vermin', there will ALWAYS be a descriptor before it.

If you absolutely insist I remove vermin from the list then I can, but I think you're being a little too fussy about this.

File: code/modules/mob/living/carbon/alien/diona/diona_nymph.dm [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 //Diona time variables, these differ slightly between a gestalt and a nymph. All values are times in seconds
 /mob/living/carbon/alien/diona
+	desc = "The most basic form of dionea."
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/alien/diona/diona_nymph.dm]:
```suggestion
	desc = "The most basic form of diona."
```

**Comment by Geevies** [comment]
@MarinaGryphon [descriptor] vermin sounds absolutely fine to me, do you still have any issues with it?

File: code/modules/mob/living/simple_animal/friendly/rat.dm [hunk 5]
```diff
@@ -100,6 +100,32 @@
...
+	if (!input)
+		return
+	if (length(input) > MAX_NAME_LEN)
+		to_chat(M, "That name's too long, maximum is [MAX_NAME_LEN] characters.")
+
+	if (!name_pattern.Find(input))
+		to_chat(M, bad_name_message)
+		return
+	if (QDELETED(src) || stat)
+		to_chat(M, "Mob no longer exists or is dead.")
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/friendly/rat.dm]: This probably needs a better description. Maybe one that fits more IC.

File: code/modules/mob/living/simple_animal/friendly/rat.dm [hunk 6]
```diff
@@ -100,6 +100,32 @@
 			new_pixely = Clamp(new_pixely, -4, 14)
 			animate(src, pixel_y = new_pixely, time = 0.5)
 
+/mob/living/proc/rename_self_helper(mob/M, regex/name_pattern, name_message, bad_name_message)
+	var/input = input(M, name_message)
+	if (!input)
+		return
+	if (length(input) > MAX_NAME_LEN)
+		to_chat(M, "That name's too long, maximum is [MAX_NAME_LEN] characters.")
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/friendly/rat.dm]: Span classes.

File: code/modules/mob/living/simple_animal/friendly/rat.dm (hunk 5 above)
Comment (NonQueueingMatt) [reply code/modules/mob/living/simple_animal/friendly/rat.dm]: Also add a span class.

**Comment by MarinaGryphon** [comment]
> plural noun: vermin
wild animals that are believed to be harmful to crops, farm animals, or game, or that carry disease, e.g., rodents.
parasitic worms or insects.
people perceived as despicable and as causing problems for the rest of society.
"the vermin who ransacked her house"

Vermin is incorrect. It would be like calling a fox a canine as a specific noun. Yes, they're canines, but that doesn't make canine the right word to describe them. Is it a dog? A wolf? A jackal? A fox? Vermin is similarly a categorical term.

**Comment by Kaedwuff** [comment]
> > plural noun: vermin
> > wild animals that are believed to be harmful to crops, farm animals, or game, or that carry disease, e.g., rodents.
> > parasitic worms or insects.
> > people perceived as despicable and as causing problems for the rest of society.
> > "the vermin who ransacked her house"
> 
> Vermin is incorrect. It would be like calling a fox a canine as a specific noun. Yes, they're canines, but that doesn't make canine the right word to describe them. Is it a dog? A wolf? A jackal? A fox? Vermin is similarly a categorical term.

Lack of specificity does not make something incorrect. There are no other 'vermin' by which you can confuse them with.

File: code/modules/mob/living/carbon/alien/alien.dm [hunk 7]
```diff
@@ -27,7 +30,7 @@
 	verbs += /mob/living/proc/ventcrawl
 	verbs += /mob/living/proc/hide
 
-	name = "[initial(name)] ([rand(1, 1000)])"
+	name = " ([get_descriptor()] [initial(name)]"
```
Comment (Geevies) [inline code/modules/mob/living/carbon/alien/alien.dm]: Is the space there meant to be there?

File: code/modules/mob/living/carbon/human/species/station/monkey.dm [hunk 8]
```diff
@@ -135,7 +134,7 @@
 	inherent_verbs = list(
 		/mob/living/carbon/human/proc/bugbite
 		)
-
+
```
Comment (Geevies) [inline code/modules/mob/living/carbon/human/species/station/monkey.dm]:
```suggestion

```

File: code/modules/mob/living/carbon/alien/alien.dm (hunk 7 above)
Comment (Kaedwuff) [reply code/modules/mob/living/carbon/alien/alien.dm]: fixed

File: code/modules/mob/living/carbon/alien/diona/diona_nymph.dm (hunk 4 above)
Comment (Kaedwuff) [reply code/modules/mob/living/carbon/alien/diona/diona_nymph.dm]: Is there anything else you think needs fixing?

**Pull request closed without merging**